### ISSUE TITLE
Input DTO not respected by OpenAPI Docs

### ISSUE THREAD

**Issue by BTL-UG** [api-platform/api-platform#1951]
**API Platform version(s) affected**: 2.6.4

**Description**  
If I use a DTO for **input**, it is not respected by the generated OpenAPO Doc. For the request body of a PUT request for example, the docs seem to be using the **output** DTO (or none, if there isn't an output DTO) instead of the input DTO.

**How to reproduce**  
Just create a simple entity, a simple DTO for input and one for output and have a look at the docs of the PUT operation.

**Additional Context**  
```
<?php
namespace App\Entity;
use ApiPlatform\Core\Annotation\ApiProperty;
use ApiPlatform\Core\Annotation\ApiResource;
use App\Dto\TestInput;
use App\Dto\TestOutput;
#[ApiResource(
    input: TestInput::class,
    output: TestOutput::class
)]
class Test
{
    #[ApiProperty(identifier: true)]
    public $id;
}
```

and the DTOs

```
<?php

namespace App\Dto;

final class TestInput {
    public string $testIn;
}

<?php

namespace App\Dto;

final class TestOutput {

    public string $testOut;
}

```

![image](https://user-images.githubusercontent.com/45096988/123645036-40a76080-d826-11eb-9a95-46b52c5faadf.png)

**Comment by alanpoulain** [comment]
This problem should be solved in `2.6.5`. Please update `api-platform/core`.

**Comment by BTL-UG** [comment]
Confirmed. Great, thanks! And by the way, impressive response time. :-)
